include/ideep/tensor.hpp

```diff
@@ -56,6 +56,16 @@ class tensor : public memory {
       set_g(1);
     }
 
+    // Construct a memory descriptor from a binary blob.
+    // The API is available since oneDNN v3.4.1
+    desc(const std::vector<uint8_t> &blob) : memory::desc(get_dnnl_blob(blob)) {
```

Comment:
For these new onednn APIs, shall we use macros to guard them?

Reply:
Macros will be used in the PyTorch when invoking these new ideep APIs, for example
(`2` in the `IDEEP_PREREQ(3, 4, 1, 2)` is the `IDEEP_VERSION_REVISION` while `3, 4, 1` is the oneDNN version):
```cpp
#if IDEEP_PREREQ(3, 4, 1, 2)
  std::vector<uint8_t> serialized_wei_desc = packed_w_desc.get_blob();
#else
      TORCH_CHECK(false, "Unexpected IDeep version to do weight serialization.");
#endif
```
This is to address the situation where we add these new ideep APIs calls in PyTorch but Meta doesn't upgrade the ideep version internally.

Do we also need to consider the situation where ideep is updated but the mkldnn submodule is not updated?

Reply:
Oh, that's good point. No, we don't need to handle the misaligned ideep vs. onednn versions.